addons/hr_recruitment/wizard/applicant_refuse_reason.py

```diff
@@ -9,6 +10,43 @@ class ApplicantGetRefuseReason(models.TransientModel):
 
     refuse_reason_id = fields.Many2one('hr.applicant.refuse.reason', 'Refuse Reason')
     applicant_ids = fields.Many2many('hr.applicant')
+    send_mail = fields.Boolean("Send Email", compute='_compute_send_mail', store=True, readonly=False)
+    template_id = fields.Many2one('mail.template', string='Email Template',
+        compute='_compute_send_mail', store=True, readonly=False,
+        domain="[('model', '=', 'hr.applicant')]")
+    applicant_without_email = fields.Text(compute='_compute_applicant_without_email',
+        string='Applicant(s) not having email')
+
+    @api.depends('refuse_reason_id')
+    def _compute_send_mail(self):
+        for wizard in self:
+            template = wizard.refuse_reason_id.template_id
+            wizard.send_mail = bool(template)
+            wizard.template_id = template
+
+    @api.depends('applicant_ids', 'send_mail')
+    def _compute_applicant_without_email(self):
+        for wizard in self:
+            applicants = wizard.applicant_ids.filtered(lambda x: not x.email_from and not x.partner_id.email)
+            if applicants and len(wizard.applicant_ids) > 1 and self.send_mail:
```

Comment:
Why do you need `len(wizard.applicant_ids) > 1`? I think you can remove this.

Reply:
@jdoutreloux ,
This warning will only appears when we have selected multiple applicants from the list view and some of them does not have the email.
for the single application(from form view) we does not require the warning as it will raise the warning if applicant does not have email.

Thanks